### PR TITLE
Toniof 868 opensearch

### PR DESCRIPTION
**Pull request by TonioF**
Use new opensearch server as defaut odp data store.
Solves #868 .

### CODE REVIEW

File: cate/ds/__init__.py [hunk 1]
```diff
@@ -48,5 +48,8 @@ def cate_init():
     from .esa_cci_odp import set_default_data_store
     set_default_data_store()
 
+    from .esa_cci_odp_os import add_data_store
+    add_data_store()
```
Comment (forman) [inline cate/ds/__init__.py]: Should be default.

File: cate/ds/__init__.py [hunk 2]
```diff
@@ -48,5 +48,8 @@ def cate_init():
     from .esa_cci_odp import set_default_data_store
     set_default_data_store()
```
Comment (forman) [inline cate/ds/__init__.py]: Only use for testing, e.g. set a flag `USE_ODP_LEGACY_DATA_STORE = True` or put in config

File: cate/ds/__init__.py [hunk 3]
```diff
@@ -48,5 +48,8 @@ def cate_init():
     from .esa_cci_odp import set_default_data_store
```
Comment (forman) [inline cate/ds/__init__.py]: raname `esa_cci_odp`   to `esa_cci_odp_legacy`

File: cate/ds/__init__.py [hunk 4]
```diff
@@ -48,5 +48,8 @@ def cate_init():
...
     set_default_data_store()
 
+    from .esa_cci_odp_os import add_data_store
```
Comment (forman) [inline cate/ds/__init__.py]: rename `esa_cci_odp_os` into `esa_cci_odp`

File: cate/ds/esa_cci_odp_os.py [hunk 5]
```diff
@@ -0,0 +1,1376 @@
...
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+"""
+Description
+===========
+
+This plugin module adds the ESA CCI Open Data Portal's (ODP) OpenSearch service to
```
Comment (forman) [inline cate/ds/esa_cci_odp_os.py]:
```suggestion
This plugin module adds the ESA CCI Open Data Portal's (ODP) service to
```

File: cate/ds/esa_cci_odp_os.py [hunk 6]
```diff
@@ -0,0 +1,1376 @@
...
+the data store registry.
```
Comment (forman) [inline cate/ds/esa_cci_odp_os.py]:
```suggestion
the data store registry. As of April 2020, the ODP service povided a OpenSearch-compatible catalogue,
which is utilised in this implementation.
```

File: cate/ds/esa_cci_odp_os.py [hunk 7]
```diff
@@ -0,0 +1,1376 @@
...
+    def _get_urls_list(files_description_list, protocol) -> Sequence[str]:
+        """
+        Returns urls list extracted from reference esgf specific files description json list
+        :param files_description_list:
+        :param protocol:
+        :return:
+        """
+        return [file_rec[4][protocol].replace('.html', '') for file_rec in files_description_list]
+
+    def _make_local(self,
```
Comment (forman) [inline cate/ds/esa_cci_odp_os.py]: This method is way too large. Split into smaller peaces, so this code becomes human-readable and testable.

Comment (forman) [reply cate/ds/esa_cci_odp_os.py]: Why is this here at all, when there is `make_local(self, ...)` too?

File: cate/ds/esa_cci_odp_os.py [hunk 8]
```diff
@@ -0,0 +1,1376 @@
...
+                                _LOG.info(f"Downloading {actual_url} to {dataset_file}")
+                                urllib.request.urlretrieve(actual_url, filename=dataset_file, reporthook=reporthook)
+                            file_number += 1
+                            local_ds.add_dataset(os.path.join(local_id, filename), (coverage_from, coverage_to))
+
+                            if do_update_of_verified_time_coverage_start_once:
+                                verified_time_coverage_start = coverage_from
+                                do_update_of_verified_time_coverage_start_once = False
+                            verified_time_coverage_end = coverage_to
+        except HTTPError as e:
```
Comment (forman) [inline cate/ds/esa_cci_odp_os.py]: Way to large try/catch scope!

File: cate/ds/__init__.py (hunk 4 above)
Comment (TonioF) [reply cate/ds/__init__.py]: Okay. I will consequently add "legacy" from terms that refer to the old data stores and remove "opensearch" to ones that refer to the new one - up to that the identifiers change.

**Comment by TonioF** [comment]
I have addressed the issues from the first iteration.

File: cate/ds/__init__.py [hunk 9]
```diff
@@ -44,9 +44,14 @@ def cate_init():
     # Plugin initializer.
     # Sets the default data store.
 
-    # from .esa_cci_ftp import set_default_data_store
     from .esa_cci_odp import set_default_data_store
     set_default_data_store()
 
+    # from .esa_cci_ftp import set_default_data_store
+    import os
+    if(os.environ.get('USE_ODP_LEGACY_DATA_STORE', False)):
```
Comment (forman) [inline cate/ds/__init__.py]:
```suggestion
    if os.environ.get('USE_ODP_LEGACY_DATA_STORE', False):
```

**Pull request merged**